src/context/TodosContext.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { ErrorType, FilterType, Todo } from '../types';
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from '../api/todos';
+import { useAuthContext } from './AuthContext';
+
+type Props = {
+  children: ReactNode;
+};
+
+type TodoProviderType = {
+  loading: number[];
+  setLoading: (loading: number[]) => void
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  errors: ErrorType | null;
+  setErrors: (error: ErrorType | null) => void;
+  filter: FilterType;
+  setFilter: (filter: FilterType) => void;
+  tempTodo: Todo | null;
+  setTempTodo: (todo: Todo) => void;
+  filteredTodos: Todo[];
+  inProgress: number;
+  deleteTodoFromServer: (todoId: number) => void;
+  deleteCompletedTodos: () => void;
+  addTodoToServer: (todo: Omit<Todo, 'id'>) => Promise<boolean>;
+  updateTodoOnServer: (id:number, todo: Partial<Todo>) => Promise<void>;
+  updateAllTodoOnServer: (status: boolean) => Promise<void>;
+};
+
+const TodoContext = createContext<TodoProviderType>({
+  loading: [],
+  setLoading: () => [],
+  todos: [],
+  setTodos: () => {},
+  errors: null,
+  setErrors: () => {},
+  filter: FilterType.All,
+  setFilter: () => {},
+  filteredTodos: [],
+  tempTodo: null,
+  setTempTodo: () => {},
+  inProgress: 0,
+  deleteTodoFromServer: () => {},
+  deleteCompletedTodos: () => {},
+  addTodoToServer: async () => false,
+  updateTodoOnServer: async () => {},
+  updateAllTodoOnServer: async () => {},
+});
+
+export const TodosProvider: FC<Props> = ({ children }) => {
+  const [loading, setLoading] = useState<number[]>([]);
```

Comment:
i think better name will be loadingTodos

Reply:
done